stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt

```diff
@@ -23,348 +22,446 @@ public interface UserEvent {
     public val user: User
 }
 
-public data class ChannelCreatedEvent(
-    override val type: String,
-    @SerializedName("created_at") override val createdAt: Date,
-    override val cid: String,
-    @SerializedName("channel_type") val channelType: String,
-    @SerializedName("channel_id") val channelId: String,
-    override val user: User,
-    val message: Message?,
-    val channel: Channel,
-) : CidEvent(), UserEvent
+public interface HasChannel {
+    public val cid: String
+    public val channelType: String
+    public val channelId: String
+}
+
+public interface HasMessage {
+    public val message: Message
+}
+
+public interface HasReaction {
+    public val reaction: Reaction
+}
 
+public interface HasMember {
+    public val member: Member
+}
+
+public interface HasOwnUser {
+    public val me: User
+}
+
+public interface HasWatcherCount {
+    public val watcherCount: Int
+}
+
+public interface HasUnreadCounts {
+    public val totalUnreadCount: Int?
+    public val unreadChannels: Int?
+}
```

Comment:
This interface looks weird. It said that has unreadCounts but the values are nullable

Reply:
I'm checking it... According to the spec these fields are not present in "message.new" and "notification.message_new" events, but I see them coming from the server with these events.